### PR TITLE
ACOM updates

### PR DESCRIPTION
**Pull request by cophus**
This PR will add fully automated strain mapping to ACOM.  Also numerous other improvements and bugfixes to ACOM.

### CODE REVIEW

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 1]
```diff
@@ -1430,6 +1433,132 @@ def match_single_pattern(
...
+    strain_map.slices['mask'] = 1
+
+    # init values
+    if corr_kernel_size is None:
+        corr_kernel_size = self.orientation_kernel_size
+    radius_max_2 = corr_kernel_size**2
+
+    # Loop over all probe positions
+    for rx, ry in tqdmnd(
+        # 10,10,
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_ACOM.py:1482]: Can we clean these up?

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 2]
```diff
@@ -1430,6 +1433,132 @@ def match_single_pattern(
...
+        desc="Calculating strains",
+        unit=" PointList",
+        disable=not progress_bar,
+        ):
+        # Get bragg peaks from experiment and reference
+        p = bragg_peaks_array.get_pointlist(rx,ry)
+
+        if p.data.shape[0] >= min_num_peaks:
+            p_ref = self.generate_diffraction_pattern(
+                orientation_map.get_orientation(rx,ry),
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_ACOM.py:1497]: I think we should also let the user specify the other relevant arguments to `generate_diffraction_pattern`, i.e. `tol_excitation_error_mult` and `k_max`

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 3]
```diff
@@ -1430,6 +1433,132 @@ def match_single_pattern(
...
+            else:
+                strain_map.slices['theta'][rx,ry] \
+                    -= (orientation_map.angles[rx,ry,0,0] \
+                    + orientation_map.angles[rx,ry,0,2])
+
+        else:
+            strain_map.slices['mask'][rx,ry] = 0
+
+    if rotation_range is not None:
+        print(1)
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_ACOM.py:1554]: Remove

**Review by sezelt — COMMENTED** [review]
I haven't tested it out yet but it looks good to me, I just added some comments on places to clean up debugging stuff. Also, suggesting that let the user specify all of the arguments that get passed to `generate_diffraction_pattern`

File: py4DSTEM/process/diffraction/crystal_ACOM.py (hunk 2 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_ACOM.py:1643]: Done and done!  All default options copied over, with exact same argument names.

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 4]
```diff
@@ -27,6 +36,7 @@ def orientation_plan(
     fiber_axis=None,
     fiber_angles=None,
     figsize: Union[list, tuple, np.ndarray] = (6, 6),
+    CUDA: bool = True,
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_ACOM.py:39]: Don't make this True by default!

**Review by sezelt — CHANGES_REQUESTED** [review]
I think this just needs the two little tweaks I flagged: make CUDA off by default, and propagate `min_num_peaks`

File: py4DSTEM/process/diffraction/crystal_ACOM.py (hunk 4 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_ACOM.py]: Good point!  I got too excited about adding CUDA support.

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 5]
```diff
@@ -742,24 +744,28 @@ def orientation_plan(
...
     # Fourier domain along angular axis
-    self.orientation_ref = np.conj(np.fft.fft(self.orientation_ref))
+    if self.CUDA:
+        self.orientation_ref = cp.asarray(self.orientation_ref)
+        self.orientation_ref = cp.conj(cp.fft.fft(self.orientation_ref))
+    else:
+        self.orientation_ref = np.conj(np.fft.fft(self.orientation_ref))
 
 
 def match_orientations(
```
[the remark this answers is not in the document]
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_ACOM.py:763]: Good catch, fixed.

**Review by bsavitzky — APPROVED** [review]

**Comment by bsavitzky** [comment]
I'm merging this over the block from Steve's unresolved request for edits, because those changes have been made and I'm unsure why they're still listed as unresolved!

**Pull request merged**